extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -157,12 +167,10 @@ export class PublishOptionsDialog {
 	* Reset button click, resets all the options selection
 	*/
 	private async reset(): Promise<void> {
+		this.resetClicked = true;
 		const result = await this.publish.getDefaultDeploymentOptions();
 		this.optionsModel.deploymentOptions = result;
```

Comment:
Do you really need to get the default options and set them here? As discussed the options themselves shouldn't change - and the model already has the "original" state of the options (in `deploymentOptions`) so couldn't you just call `setOptionsToValueNameLookup` and reset the values there back to their defaults?

Reply:
If we want to maintain the defaultOptions, we can do either
1. Have a deep copy at parent level or dialog level, and use that for reset, by updating the original one as usual, or
2. not updating the parent level defaultOptions at any time until we click the publish/generateScript button!